eslint.config.js

```diff
@@ -50,6 +50,45 @@ export default tseslint.config(
     extends: [reactConfig, pluginQuery.configs['flat/recommended']],
     rules: {
       '@typescript-eslint/only-throw-error': 'off',
+      'no-duplicate-imports': 'off',
+      '@typescript-eslint/no-duplicate-imports': ['error'],
+      // Component and hook definition rules
+      'react/function-component-definition': ['error', {
+        'namedComponents': 'function-declaration',
+        'unnamedComponents': 'arrow-function'
+      }],
+      // TypeScript rules
+      '@typescript-eslint/consistent-type-definitions': ['error', 'interface'],
+      '@typescript-eslint/explicit-function-return-type': ['error', {
+        'allowExpressions': true
+      }],
+      '@typescript-eslint/no-floating-promises': 'error',
+      '@typescript-eslint/no-unused-vars': 'error',
+      // Code quality
+      'react/jsx-fragments': ['error', 'element'],
+      'prefer-const': 'error',
+      'eqeqeq': ['error', 'always', { 'null': 'ignore' }],
+      // Import ordering
+      'import/order': ['error', {
+        'groups': ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
+        'pathGroups': [
+          {
+            'pattern': 'react',
+            'group': 'external',
+            'position': 'before'
+          },
+          {
+            'pattern': '@leather*',
+            'group': 'external',
+            'position': 'after'
+          }
+        ],
+        'newlines-between': 'always',
+        'alphabetize': {
+          'order': 'asc',
+          'caseInsensitive': true
+        }
+      }]
     },
```

Comment:
Why have these rules been added changed?

`'no-duplicate-imports': 'off',` we want this on

Reply:
We use prettier to enforce import ordering etc

Reply:
We also define rules in our eslint package, not directly in the config. Suggest removing these

Reply:
@kyranjamie that was my bad, I was seeing complaints about duplicate imports from 
```js
import type MyType from 'foo';
import useThings from 'foo';
```

sounds like maybe it was my environment though